samples/Azure.ResourceManager.Sample/Generated/AvailabilitySetContainer.cs

```diff
@@ -437,6 +437,106 @@ public virtual AsyncPageable<GenericResource> GetAllAsGenericResourcesAsync(stri
             }
         }
 
+        /// <summary> Delete an availability set. </summary>
+        /// <param name="availabilitySetName"> The name of the availability set. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="availabilitySetName"/> is null. </exception>
+        public async virtual Task<Response> DeleteAsync(string availabilitySetName, CancellationToken cancellationToken = default)
+        {
+            if (availabilitySetName == null)
+            {
+                throw new ArgumentNullException(nameof(availabilitySetName));
+            }
+
+            using var scope = _clientDiagnostics.CreateScope("AvailabilitySetContainer.Delete");
+            scope.Start();
+            try
+            {
+                var operation = await StartDeleteAsync(availabilitySetName, cancellationToken).ConfigureAwait(false);
+                return await operation.WaitForCompletionResponseAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
```

Comment:
Why are we generating one delete method in two classes([Resource class](https://github.com/Azure/autorest.csharp/blob/456c9e62c2a7519862473468abeedbc5189e6e08/samples/Azure.ResourceManager.Sample/Generated/AvailabilitySet.cs#L131-L165) and ResourceContainer class)?

Reply:
Yes. That's what discussed in [ADO 5998](). We found that if users want to delete a resource by name, they should be able to delete without getting the resource first. Adding `Delete` to the container will save one unnecessary HTTP Get request in some cases.

This is not a preview blocker. I'll hold on this until other changes for preview gets in first.